h/accounts/views.py

```diff
@@ -565,6 +565,36 @@ def _user_notifications(self):
             self.request.authenticated_userid)
 
 
+@view_defaults(route_name='profile_developer',
+               renderer='h:templates/accounts/developer.html.jinja2',
+               effective_principals=security.Authenticated)
+class DeveloperController(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @view_config(request_method='GET')
+    def get(self):
+        """Render the developer page, including the form."""
+        token = models.Token.get(self.request.authenticated_userid)
+        if token:
+            return {'token': token}
+        else:
+            return {}
+
+    @view_config(request_method='POST')
+    def post(self):
+        """Process notifications POST data."""
+        old_token = models.Token.get(self.request.authenticated_userid)
```

Comment:
It would be clearer I think to have a `.refresh()` method on the token model which generates a new token. That way there's no need to faff with the database session.